cmd/soroban-rpc/internal/integrationtest/infrastructure/client.go

```diff
@@ -0,0 +1,196 @@
+	assert.NoError(t, err)
+
+	assert.Equal(t, expectedHashHex, result.Hash)
+	if !assert.Equal(t, stellarcore.TXStatusPending, result.Status) {
+		var txResult xdr.TransactionResult
+		err := xdr.SafeUnmarshalBase64(result.ErrorResultXDR, &txResult)
+		assert.NoError(t, err)
+		t.Logf("error: %#v\n", txResult)
+	}
+	assert.NotZero(t, result.LatestLedger)
+	assert.NotZero(t, result.LatestLedgerCloseTime)
+
+	response := getTransaction(t, client, expectedHashHex)
+	if !assert.Equal(t, methods.TransactionStatusSuccess, response.Status) {
+		var txResult xdr.TransactionResult
+		assert.NoError(t, xdr.SafeUnmarshalBase64(response.ResultXdr, &txResult))
+		t.Logf("error: %#v\n", txResult)
+
+		var txMeta xdr.TransactionMeta
+		assert.NoError(t, xdr.SafeUnmarshalBase64(response.ResultMetaXdr, &txMeta))
+
+		if txMeta.V == 3 && txMeta.V3.SorobanMeta != nil {
+			if len(txMeta.V3.SorobanMeta.Events) > 0 {
+				t.Log("Contract events:")
+				for i, e := range txMeta.V3.SorobanMeta.Events {
+					t.Logf("  %d: %s\n", i, e)
+				}
+			}
+
+			if len(txMeta.V3.SorobanMeta.DiagnosticEvents) > 0 {
+				t.Log("Diagnostic events:")
+				for i, d := range txMeta.V3.SorobanMeta.DiagnosticEvents {
+					t.Logf("  %d: %s\n", i, d)
+				}
+			}
+		}
+	}
+
+	require.NotNil(t, response.ResultXdr)
+	assert.Greater(t, response.Ledger, result.LatestLedger)
+	assert.Greater(t, response.LedgerCloseTime, result.LatestLedgerCloseTime)
+	assert.GreaterOrEqual(t, response.LatestLedger, response.Ledger)
+	assert.GreaterOrEqual(t, response.LatestLedgerCloseTime, response.LedgerCloseTime)
+	return response
+}
+
+func SimulateTransactionFromTxParams(t *testing.T, client *Client, params txnbuild.TransactionParams) methods.SimulateTransactionResponse {
+	savedAutoIncrement := params.IncrementSequenceNum
+	params.IncrementSequenceNum = false
+	tx, err := txnbuild.NewTransaction(params)
+	require.NoError(t, err)
+	params.IncrementSequenceNum = savedAutoIncrement
+	txB64, err := tx.Base64()
+	require.NoError(t, err)
+	request := methods.SimulateTransactionRequest{Transaction: txB64}
+	var response methods.SimulateTransactionResponse
+	err = client.CallResult(context.Background(), "simulateTransaction", request, &response)
+	require.NoError(t, err)
+	return response
+}
+
+func PreflightTransactionParamsLocally(t *testing.T, params txnbuild.TransactionParams, response methods.SimulateTransactionResponse) txnbuild.TransactionParams {
+	if !assert.Empty(t, response.Error) {
+		fmt.Println(response.Error)
+	}
+	var transactionData xdr.SorobanTransactionData
+	err := xdr.SafeUnmarshalBase64(response.TransactionData, &transactionData)
+	require.NoError(t, err)
+
+	op := params.Operations[0]
+	switch v := op.(type) {
+	case *txnbuild.InvokeHostFunction:
+		require.Len(t, response.Results, 1)
+		v.Ext = xdr.TransactionExt{
+			V:           1,
+			SorobanData: &transactionData,
+		}
+		var auth []xdr.SorobanAuthorizationEntry
+		for _, b64 := range response.Results[0].Auth {
```

Comment:
totally separate but we really ought to just write a Go client for Soroban RPC at this point

Reply:
Yes, I totally agree, I suggested it to @mollykarcher a while ago and I recall she mentioned it wasn't a priority (although refactoring the whole integration test infrastructure probably isn't either).

Now seriously, I think it makes sense to create an internal client at least. Exposing it externally carries a higher maintenance burden.

Reply:
Yeah internal makes a lot of sense so that we can make whatever changes we want.